src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
I'm not so sure.  You could consider (modulename, targetname) more specific than just targetname.  On the other hand, actively registering a function feels more specific than looking up something that is installed.  I think we need to standardize on one way for our stuff, and leave the other option for easy customization.